WTG.Analyzers/Analyzers/PointlessOverride/PointlessOverrideAnalyzer.cs

```diff
@@ -0,0 +1,362 @@
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using WTG.Analyzers.Utils;
+
+namespace WTG.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	class PointlessOverrideAnalyzer : DiagnosticAnalyzer
+	{
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(
+			Rules.RemovePointlessOverridesRule);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction(CompilationStart);
+		}
+
+		static void CompilationStart(CompilationStartAnalysisContext context)
+		{
+			var cache = new FileDetailCache();
+
+			context.RegisterSyntaxNodeAction(
+				c => Analyze(c, cache),
+				SyntaxKind.MethodDeclaration,
+				SyntaxKind.PropertyDeclaration,
+				SyntaxKind.IndexerDeclaration,
+				SyntaxKind.EventDeclaration);
+		}
+
+		static void Analyze(SyntaxNodeAnalysisContext context, FileDetailCache cache)
+		{
+			if (cache.IsGenerated(context.SemanticModel.SyntaxTree, context.CancellationToken))
+			{
+				return;
+			}
+
+			var node = (CSharpSyntaxNode)context.Node;
+
+			if (IsOverride(node) && !node.Accept(IsMeaningfulVisitor.Instance))
+			{
+				context.ReportDiagnostic(Rules.CreateRemovePointlessOverridesDiagnostic(node.GetLocation()));
+			}
+		}
+
+		static bool IsOverride(CSharpSyntaxNode node)
+		{
+			var modifiers = node.Accept(ModifierExtractionVisitor.Instance);
+
+			foreach (var modifier in modifiers)
+			{
+				if (modifier.IsKind(SyntaxKind.OverrideKeyword))
+				{
+					return true;
+				}
+			}
+
+			return false;
+		}
+
+		static bool IsBaseMemberAccess(ExpressionSyntax node)
+			=> node != null
+				&& node.IsKind(SyntaxKind.SimpleMemberAccessExpression)
+				&& IsBaseMemberAccess((MemberAccessExpressionSyntax)node);
+
+		static bool IsBaseMemberAccess(MemberAccessExpressionSyntax node)
+			=> node.Expression != null && node.Expression.IsKind(SyntaxKind.BaseExpression);
```

Comment:
I didn't use `?.` because I would then still have to convert the `bool?` to a `bool`. Instead, I should probably remove the null check, since I noticed IsKind will return false when called on a null node anyway.